client/src/Components/SliderArea.jsx

```diff
@@ -0,0 +1,305 @@
+      label: "3.9",
+    },
+    {
+      value: 6.8,
+      label: "6.8",
+    },
+    {
+      value: 9.7,
+      label: "9.7",
+    },
+    {
+      value: 12.6,
+      label: "12.6",
+    },
+    {
+      value: 15.5,
+      label: "15.5",
+    },
+    {
+      value: 18.4,
+      label: "18.4",
+    },
+    {
+      value: 21.3,
+      label: "21.3",
+    },
+    {
+      value: 24.2,
+      label: "24.2",
+    },
+    {
+      value: 27.1,
+      label: "27.1",
+    },
+    {
+      value: 30,
+      label: "30",
+    },
+  ];
+  
+  const delayArray = [
+    {
+      value: 1,
+      label: "1",
+    },
+    {
+      value: 13,
+      label: "13",
+    },
+    {
+      value: 25,
+      label: "25",
+    },
+    {
+      value: 37,
+      label: "37",
+    },
+    {
+      value: 49,
+      label: "49",
+    },
+    {
+      value: 61,
+      label: "61",
+    },
+    {
+      value: 72,
+      label: "72",
+    },
+    {
+      value: 84,
+      label: "84",
+    },
+    {
+      value: 96,
+      label: "96",
+    },
+    {
+      value: 108,
+      label: "108",
+    },
+    {
+      value: 120,
+      label: "120",
+    },
+  ];
+  
+  const labelArr = [miArray, ipArray, errArray, delayArray];
+  
+  const titleArr = [
+    "Monthly Investment (Rs.)",
+    "Investment Period (in years)",
+    "Expected Rate of Return (%p.a)",
+    "Delay in Staring SIP (in months)",
+  ];
+  
+  function valuetext(value) {
+    return `${value}`;
+  }
+  
+  const Input2 = styled(MuiInput)`
+    width: 60px;
+  `;
+  
+
+  const handleSliderChange = (event, newValue) => {
+    props.changeValues(props.index, newValue);
+    setInputVal(newValue);
+  };
+
+  const handleInputChange = (event) => {
+
+    //FrontEnd validations
+
+    let val = event.target.value;
+    if (val.startsWith("-")) {
+      const errField =
+        document.getElementsByClassName("err-field")[props.index];
+      errField.style.display = "block";
+      setTimeout(() => {
+        errField.style.display = "none";
+      }, 1500);
+      setInputVal(val);
+      props.changeValues(props.index, props.mn);
+      return;
+    }
+
+    if (Number(val) < props.mn) {
+      setInputVal(val);
+      props.changeValues(props.index, props.mn);
+      return;
+    }
+
+    if (Number(val) > props.mx) {
+      setInputVal(val);
+      props.changeValues(props.index, props.mx);
+      return;
+    }
+
+    setInputVal(val);
+    props.changeValues(props.index, val === "" ? props.mn : Number(val));
+  };
+
+  const handleBlur = (event) => {
+    let val = event.target.value;
+
+    if (val === "-") {
+      props.changeValues(props.index, props.mn);
+      setInputVal(props.mn);
+      return;
+    }
+
+    val = Number(val);
+
+    if (val < props.mn) {
+      props.changeValues(props.index, props.mn);
+      setInputVal(props.mn);
+      return;
+    }
+    if (val > props.mx) {
+      props.changeValues(props.index, props.mx);
+      setInputVal(props.mx);
+      return;
+    }
+  };
```

Comment:
write all these , validations related functions in optimized way after discussing